### DISCUSSION TITLE
How to make the Mojolicious configuration available to Mojo::Base objects even outside the application?

### DISCUSSION THREAD

**Discussion by fluca1978** [mojolicious/mojo#2182] [General]
I'm developing a bunch of classes that must be able to perform tasks within the application or outside of it, loading the application configuration (`NotYAMLConf`). What is the best way to achieve this?
I mean, I would like the object to have the configuration injected once it is instantiated, so that it can do `$self->config` as within a normal controller.

**Comment by guest20** [comment]
`Mojolicious::Command`s have the config available, or is that not far enough"outside" the app?

**Comment by Grinnz** [comment]
Commands are the normal way to do this. they will run application startup and have anything available to the application itself, including config and helpers.

**Comment by fluca1978** [comment]
A command is an excellent idea! Thanks.
